File: cpp/136-SingleNumber.cpp
// 2022-4-7
// Runtime 16ms(65.7%)
// Memory 16.5mb(28.2%)

class Solution {
public:
    int singleNumber(vector<int>& nums) {
        if(nums.size() == 1) return nums[0];
        int ret = nums[0];
        for(int i = 1; i < nums.size(); i++) {
            ret ^= nums[i];
        }
        return ret;
    }
};

File: py3/136-SingleNumber.py
# 2022-4-7
# Runtime 1056ms(11.6%)
# Memory 16.9mb(28.5%)

class Solution:
    def singleNumber(self, nums: List[int]) -> int:
        save = list()
        for i in nums:
            if i in save: save.remove(i)
            else: save.append(i)
        return save[0]
